client/state/stats/chart-tabs/reducer.js

```diff
@@ -26,16 +26,28 @@ export function counts( state = {}, action ) {
 			const records = get( state, `${ action.siteId }.${ action.period }`, [] ).slice();
 			const recordIds = records.map( count => count[ ID ] );
 
+			let areThereChanges = false;
+
 			// Merge existing records with records from API
 			action.data.forEach( recordFromApi => {
 				const index = recordIds.indexOf( recordFromApi[ ID ] );
 				if ( index >= 0 ) {
-					records[ index ] = { ...records[ index ], ...recordFromApi };
+					const newRecords = { ...records[ index ], ...recordFromApi };
+					if ( ! isEqual( newRecords, records[ index ] ) ) {
+						areThereChanges = true;
+						records[ index ] = newRecords;
+					}
 				} else {
+					areThereChanges = true;
 					records.push( recordFromApi );
 				}
 			} );
 
+			// Avoid changing state if nothing's changed.
+			if ( ! areThereChanges ) {
+				return state;
+			}
+
```

Comment:
Opportunity for some bonus points, either in this PR or in a followup: convert the reducer to use `keyedReducer`:
```js
keyedReducer( 'siteId', keyedReducer( 'period', ( state, action ) => { /* the rest */ } ) )
```
That should simplify the inner reducer a lot!

There are many other `stats` reducers that would benefit from `keyedReducer`, too.

Reply:
Hah, there's interesting libs all over the place in Calypso!

Done. I think the rewritten reducer is correct, but please take a look when you have a chance. Two sets of eyes are better than one 😄